### ISSUE TITLE
customize objective function

### ISSUE THREAD

**Issue by fly-cfchen** [fchollet/keras#1488]
I want to build my own objective function, which need to use the `softmax` results as the input. I add a function in the `objective.py`, just follow the function `categorical_crossentropy`, and it can run. 
However, when I directly put the predict `softmax` result to the objective function, the result is different from the `model.fit` function. **I use the same weight to do the test.** 
I wonder how the `model.fit` function compute the loss, is it  different from the objective function? Or the input of the objective function is not the `softmax` results?

**Comment by fly-cfchen** [comment]
I have fixed it already. The reason is that I use a dropout layer in the network.
